src/components/button-row/button-row.js

```diff
@@ -1,55 +1,29 @@
-/* Copyright (C) 2017 Canonical Ltd. */
+/* Copyright (C) 2018 Canonical Ltd. */
 'use strict';
 
-const classNames = require('classnames');
 const PropTypes = require('prop-types');
 const React = require('react');
 
 const Button = require('../button/button');
 
 require('./_button-row.scss');
 
-class ButtonRow extends React.Component {
-  /**
-    Returns the classes for the footer based on the provided props.
-    @returns {String} The collection of class names.
-  */
-  _generateClasses() {
-    var classes = {};
-    var buttonsLength = this.props.buttons.length;
-    classes['button-row--multiple'] = buttonsLength > 0;
-    classes['button-row--count-' + buttonsLength] = true;
-    return classNames('button-row', classes);
-  }
-
-  /**
-    Creates the buttons based on the provided props.
-    @param {Array} buttons The properties of the buttons to generate.
-    @returns {Array} Collection of buttons.
-  */
-  _generateButtons(buttons) {
-    var components = [];
-    buttons.forEach(button => {
-      components.push(
-        <Button
-          action={button.action}
-          disabled={button.disabled}
-          key={button.title}
-          submit={button.submit}
-          type={button.type}
-        >
-          {button.title}
-        </Button>
-      );
-    });
-    return components;
-  }
-
-  render() {
-    var buttons = this._generateButtons(this.props.buttons);
-    return <div className={this._generateClasses()}>{buttons}</div>;
-  }
-}
+const ButtonRow = props => {
+  const buttonListItems = [];
+  const buttonsArr = props.buttons;
+  buttonsArr.forEach((button, index) => {
+    buttonListItems.push(
+      <li className="p-inline-list__item" key={index}>
+        <Button {...props}>{button.title}</Button>
```

Comment:
Are you sure we want to pass all of the `ButtonRow` props to the `Button`? Wouldn't we instead want to pass the keys from the button object as props to the button?